File: app/src/main/java/com/kelsos/mbrc/di/providers/NotificationManagerCompatProvider.java
package com.kelsos.mbrc.di.providers;

import android.content.Context;
import android.support.v4.app.NotificationManagerCompat;
import javax.inject.Inject;
import javax.inject.Provider;

public class NotificationManagerCompatProvider implements Provider<NotificationManagerCompat> {
  private Context context;

  @Inject
  public NotificationManagerCompatProvider(Context context) {
    this.context = context;
  }

  @Override
  public NotificationManagerCompat get() {
    return NotificationManagerCompat.from(context);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/di/providers/NotificationManagerCompatProvider.kt
package com.kelsos.mbrc.di.providers

import android.app.Application
import android.content.Context
import androidx.core.app.NotificationManagerCompat
import javax.inject.Inject
import javax.inject.Provider

class NotificationManagerCompatProvider
@Inject
constructor(context: Application) : Provider<NotificationManagerCompat> {
  private val context: Context

  init {
    this.context = context
  }

  override fun get(): NotificationManagerCompat {
    return NotificationManagerCompat.from(context)
  }
}
